src/app/core/services/policy-definition-builder.ts

```diff
@@ -58,24 +58,39 @@ export class PolicyDefinitionBuilder {
   private buildTimePeriodRestrictionPermissions(
     formValue: NewPolicyDialogFormValue,
   ): Permission[] {
-    const start = formValue.range!!.start!!;
-    const end = addDays(formValue.range!!.end!!, 1);
+    if (!formValue.rangeIsOpenEnded) {
```

Comment:
flip if condition so that this does not need to be negated, since both blocks look similar there's no need to have the condition negated.